applicationset/services/repo_service.go

```diff
@@ -3,40 +3,56 @@ package services
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"github.com/argoproj/argo-cd/v3/pkg/apis/application/v1alpha1"
 	"github.com/argoproj/argo-cd/v3/reposerver/apiclient"
-	"github.com/argoproj/argo-cd/v3/util/git"
+	"github.com/argoproj/argo-cd/v3/util/db"
 	"github.com/argoproj/argo-cd/v3/util/io"
 )
 
 type argoCDService struct {
 	getRepository          func(ctx context.Context, url, project string) (*v1alpha1.Repository, error)
-	storecreds             git.CredsStore
 	submoduleEnabled       bool
-	repoServerClientSet    apiclient.Clientset
 	newFileGlobbingEnabled bool
+	getGitFiles            func(ctx context.Context, req *apiclient.GitFilesRequest) (*apiclient.GitFilesResponse, error)
+	getGitDirectories      func(ctx context.Context, req *apiclient.GitDirectoriesRequest) (*apiclient.GitDirectoriesResponse, error)
```

Comment:
why not having a single attribute like the following instead of 2 functions signatures in the struct?
`repo Repos`

Reply:
@leoluz I renamed those functions to hopefully make things more clear. Does that make sense?